In [1]:
import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings('ignore')

pd.set_option('max.columns', None)

In [2]:
train = pd.read_csv('Data/training_data.csv')
train.head()

Col1      Col2      Col3      Col4      Col5      Col6      Col7  \
0  0.083383  0.103556  0.122871  0.150950  0.175772  0.201790  0.224542   
1  0.353586  0.325441  0.296272  0.275924  0.247453  0.220534  0.189593   
2  0.095419  0.115687  0.134050  0.160488  0.185254  0.211083  0.232131   
3  0.698665  0.718425  0.740934  0.772865  0.781895  0.780248  0.762148   
4  0.462267  0.496878  0.534562  0.579235  0.610549  0.637590  0.649317   

       Col8      Col9     Col10     Col11     Col12     Col13     Col14  \
0  0.260662  0.295880  0.324528  0.365533  0.391972  0.432689  0.464556   
1  0.173912  0.156294  0.135051  0.124699  0.110056  0.106561  0.097347   
2  0.270633  0.302941  0.329109  0.370945  0.393027  0.433808  0.462506   
3  0.758475  0.748144  0.720218  0.709707  0.673525  0.660023  0.639333   
4  0.682783  0.708635  0.721439  0.751087  0.746780  0.761408  0.762728   

      Col15     Col16     Col17     Col18     Col19     Col20     Col21  \
0  0.500096  0.532524  0.570259  0.592834  0.630355  0.655732  0.679630   
1  0.099621  0.097448  0.096008  0.104537  0.120322  0.134268  0.144966   
2  0.495952  0.529087  0.564644  0.586140  0.624351  0.645001  0.669031   
3  0.621777  0.602368  0.586929  0.552057  0.530111  0.493121  0.451353   
4  0.763598  0.765547  0.774494  0.756916  0.755456  0.737419  0.719213   

      Col22     Col23     Col24     Col25     Col26     Col27     Col28  \
0  0.979785  0.784710  0.772336  0.746723  0.772066  0.772647  0.767681   
1  0.166781  0.187069  0.211960  0.227449  0.261966  0.288186  0.314007   
2  0.697760  0.717937  0.733818  0.733001  0.756132  0.752672  0.746319   
3  0.416271  0.374120  0.335726  0.291067  0.269485  0.242768  0.219690   
4  0.707400  0.687186  0.659505  0.620066  0.604594  0.564978  0.530373   

      Col29     Col30     Col31     Col32     Col33     Col34     Col35  \
0  0.775952  0.770374  0.760721  0.760321  0.765592  0.724601  0.703462   
1  0.343539  0.370656  0.403981  0.443430  0.483101  0.498855  0.528519   
2  0.756885  0.753025  0.745845  0.745482  0.748884  0.708096  0.686242   
3  0.198198  0.176056  0.163021  0.149787  0.125578  0.102551  0.087596   
4  0.498909  0.460349  0.428045  0.400062  0.372236  0.327639  0.293100   

      Col36     Col37     Col38     Col39     Col40     Col41     Col42  \
0  0.679174  0.650955  0.633548  0.597261  0.587476  0.541696  0.497535   
1  0.555722  0.578098  0.612729  0.629794  0.674864  0.680873  0.682558   
2  0.666051  0.637045  0.620421  0.586692  0.574394  0.534374  0.490148   
3  0.071765  0.057493  0.059611  0.056189  0.072203  0.084392  0.104498   
4  0.262765  0.227167  0.202954  0.172204  0.152319  0.124275  0.106362   

      Col43     Col44     Col45     Col46     Col47     Col48     Col49  \
0  0.466749  0.434485  0.392753  0.366640  0.332634  0.296339  0.263667   
1  0.699986  0.713939  0.716193  0.734215  0.743951  0.730587  0.725147   
2  0.459870  0.431276  0.390070  0.367492  0.336692  0.298630  0.270139   
3  0.126836  0.150832  0.162993  0.187583  0.205303  0.224360  0.246030   
4  0.086787  0.081120  0.060583  0.063997  0.056170  0.060531  0.060321   

      Col50     Col51     Col52     Col53     Col54     Col55     Col56  \
0  0.231731  0.201444  0.176189  0.151081  0.122737  0.101760  0.090976   
1  0.714117  0.714720  0.698707  0.679689  0.658591  0.635939  0.620088   
2  0.238408  0.207696  0.187901  0.161584  0.134591  0.116075  0.101217   
3  0.269687  0.300668  0.337895  0.375798  0.412580  0.451884  0.497154   
4  0.065313  0.070086  0.089750  0.106767  0.122198  0.141988  0.173844   

      Col57     Col58     Col59     Col60     Col61     Col62     Col63  \
0  0.213345  0.090705  0.075370  0.051831  0.051550  0.060180  0.062859   
1  0.614209  0.570810  0.542166  0.512280  0.477366  0.449985  0.416656   
2  0.096110  0.077745  0.080501  0.068431  0.069035  0.072498  0.074282   
3  0.548821  0.563481  0.587495  0.608713  0.621066  0.641033  0.657623   
4  0.206481  0.228812  0

In [3]:
y_train = pd.read_csv('Data/y_train.csv')
y_train.head()

output
0       5
1       1
2       4
3       3
4       4

In [4]:
test = pd.read_csv('Data/testing_data.csv')
y_test = pd.read_csv('Data/y_test.csv')

In [5]:
!pip install tensorflow

In [6]:
y_train.replace({1:0,
                2:1,
                3:2,
                4:3,
                5:4}, inplace = True)

In [7]:
import tensorflow as tf
from tensorflow import keras

In [21]:
model = keras.Sequential([
    keras.layers.Dense(128, input_dim = 128, activation = 'relu'),
    keras.layers.Dense(64, activation = 'relu'),
    keras.layers.Dropout(0.2),
    keras.layers.Dense(32, activation = 'relu'),
    keras.layers.Dense(16, activation = 'relu'),
    keras.layers.Dense(5, activation = 'softmax')
])

In [22]:
model.compile(optimizer = 'adam',
             loss = 'sparse_categorical_crossentropy',
             metrics = ['accuracy'])

In [23]:
model.fit(train, y_train, epochs = 100)

Epoch 1/100
252/252 [==============================] - 0s 716us/step - loss: 1.4727 - accuracy: 0.3018
Epoch 2/100
252/252 [==============================] - 0s 712us/step - loss: 1.1600 - accuracy: 0.4188
Epoch 3/100
252/252 [==============================] - 0s 719us/step - loss: 1.1422 - accuracy: 0.4211
Epoch 4/100
252/252 [==============================] - 0s 735us/step - loss: 1.1407 - accuracy: 0.4242
Epoch 5/100
252/252 [==============================] - 0s 722us/step - loss: 1.1379 - accuracy: 0.4283
Epoch 6/100
252/252 [==============================] - 0s 727us/step - loss: 1.1400 - accuracy: 0.4225
Epoch 7/100
252/252 [==============================] - 0s 717us/step - loss: 1.1372 - accuracy: 0.4272
Epoch 8/100
252/252 [==============================] - 0s 716us/step - loss: 1.1364 - accuracy: 0.4305
Epoch 9/100
252/252 [==============================] - 0s 743us/step - loss: 1.1431 - accuracy: 0.4272
Epoch 10/100
252/252 [==============================] - 0s 736us/step - l

252/252 [==============================] - 0s 656us/step - loss: 0.3406 - accuracy: 0.8276
Epoch 80/100
252/252 [==============================] - 0s 588us/step - loss: 0.3612 - accuracy: 0.8159
Epoch 81/100
252/252 [==============================] - 0s 640us/step - loss: 0.3347 - accuracy: 0.8308
Epoch 82/100
252/252 [==============================] - 0s 671us/step - loss: 0.3944 - accuracy: 0.7996
Epoch 83/100
252/252 [==============================] - 0s 645us/step - loss: 0.3330 - accuracy: 0.8384
Epoch 84/100
252/252 [==============================] - 0s 637us/step - loss: 0.3104 - accuracy: 0.8477
Epoch 85/100
252/252 [==============================] - 0s 656us/step - loss: 0.3354 - accuracy: 0.8561
Epoch 86/100
252/252 [==============================] - 0s 649us/step - loss: 0.2815 - accuracy: 0.8756
Epoch 87/100
252/252 [==============================] - 0s 636us/step - loss: 0.3231 - accuracy: 0.8571
Epoch 88/100
252/252 [==============================] - 0s 652us/step - loss:

In [24]:
y_test.replace({1:0,
                2:1,
                3:2,
                4:3,
                5:4}, inplace = True)

In [25]:
model.evaluate(test, y_test)

124/124 [==============================] - 0s 432us/step - loss: 0.1193 - accuracy: 0.9639


[0.11927837878465652, 0.9638888835906982]

**So our ANN model is giving an accuracy of 0.96 on the testing data**